### PR TITLE
[v15] Set the access request max duration to 14 days, add okta-requester role.

### PR DESCRIPTION
**Pull request by mdwn**
The max duration for an access request is now 14 days. A new role called okta-requester has been introduced which takes advantage of this new, expanded access request in order to improve day one Okta operations and our dogfooding effort. This will allow users requesting Okta resources to request them for longer periods of time, allowing approvers more time to approve the given access requests.

Backport #37947 to branch/v15

changelog: The maximum duration for an access request is now 14 days, the okta-requester role has been added which takes advantage of this.

### CODE REVIEW

File: constants.go [hunk 1]
```diff
@@ -668,6 +668,16 @@ const (
 	// during the setup of Connect My Computer. The prefix is followed by the name of the cluster
 	// user. See teleterm.connectmycomputer.RoleSetup.
 	ConnectMyComputerRoleNamePrefix = "connect-my-computer-"
+
+	// SystemOktaRequesterRoleName is a name of a system role that allows
```
Comment (zmb3) [inline constants.go:672]: Did you group some other unrelated changes in this PR? Based on the title I'm not sure I'd expect to see anything about Okta here.

File: lib/services/presets.go [hunk 2]
```diff
@@ -484,6 +484,70 @@ func NewPresetRequireTrustedDeviceRole() types.Role {
 	}
 }
 
+// SystemOktaAccessRoleName is the name of the system role that allows
```
Comment (zmb3) [inline lib/services/presets.go:487]: The godoc for `NewSystemOktaAccessRole` should start with `// NewSystemOktaAccessRole ...`.

Why are we discussing a role name here in a function that returns a role?

Same goes for below.

File: lib/services/presets.go [hunk 3]
```diff
@@ -685,3 +779,17 @@ func resourceBelongsToRules(rules []types.Rule, resources []string) bool {
 
 	return false
 }
+
+func updateAllowLabels(role types.Role, kind string, defaultLabels types.Labels) (bool, error) {
+	var changed bool
+	if unset, err := labelMatchersUnset(role, kind); err != nil {
+		return false, trace.Wrap(err)
+	} else if unset && len(defaultLabels) > 0 {
```
Comment (zmb3) [inline lib/services/presets.go:787]:
```suggestion
	}
	if unset && len(defaultLabels) > 0 {
```

File: constants.go (hunk 1 above)
Comment (mdwn) [reply constants.go:672]: I added these changes in based off of discussion in the dev-identity slack. I should have updated the PR description here, apologies for that. The idea was "Extend access request max duration and make it possible to use it in Okta."

File: lib/services/presets.go [hunk 4]
```diff
@@ -685,3 +779,19 @@ func resourceBelongsToRules(rules []types.Rule, resources []string) bool {
...
 	return false
 }
+
+func updateAllowLabels(role types.Role, kind string, defaultLabels types.Labels) (bool, error) {
+	var changed bool
+	unset, err := labelMatchersUnset(role, kind)
+	if err != nil {
+		return false, trace.Wrap(err)
+	}
+	if unset && len(defaultLabels) > 0 {
```
Comment (zmb3) [inline lib/services/presets.go:789]:
```suggestion
	unset, err := labelMatchersUnset(role, kind)
	if err != nil {
		return false, trace.Wrap(err)
	}
	
	var changed bool
	if unset && len(defaultLabels) > 0 {
```

**Review by zmb3 — APPROVED** [review]

**Review by smallinsky — APPROVED** [review]
